flink-connectors/flink-connector-elasticsearch-base/src/main/java/org/apache/flink/table/descriptors/ElasticsearchValidator.java

```diff
@@ -76,12 +78,18 @@ public void validate(DescriptorProperties properties) {
 		properties.validateValue(CONNECTOR_TYPE, CONNECTOR_TYPE_VALUE_ELASTICSEARCH, false);
 		validateVersion(properties);
 		validateHosts(properties);
+		validateAuth(properties);
 		validateGeneralProperties(properties);
 		validateFailureHandler(properties);
 		validateBulkFlush(properties);
 		validateConnectionProperties(properties);
 	}
 
+	private void validateAuth(DescriptorProperties properties) {
+		properties.validateString(CONNECTOR_USERNAME, true);
+		properties.validateString(CONNECTOR_PASSWORD, true);
```

Comment:
We should validate that both username and password should exist if one of them is configured.